spyder/widgets/fileswitcher.py

```diff
@@ -536,11 +589,22 @@ def setup_file_list(self, filter_text, current_path):
 
         # Sort the obtained scores and populate the list widget
         self.filtered_path = []
+        plugin = None
         for result in sorted(results):
             index = result[1]
-            text = result[-1]
             path = paths[index]
-            item = QListWidgetItem(ima.icon('FileIcon'), text)
+            icon = icons[index]
+            text = ''
+            try:
+                title = self.widgets[index][1].get_plugin_title().split(' - ')
+                if plugin != title[0]:
+                    plugin = title[0]
+                    text += '<big><b>' + plugin + '</b></big><br>'
```

Comment:
Let's add some vertical separation between the plugin title and its list of files/tabs.